code/surface_area/utils.py

```diff
@@ -0,0 +1,81 @@
+from pathlib import Path
+from typing import Iterable
+import nibabel as nib
+
+
+def find_surface_files(
+    input_dir: Path, patterns: Iterable[str] | None = None
+) -> list[Path]:
+    """
+    Find surface files matching the specified patterns.
+    """
+    if patterns is None:
+        patterns = ["*mid*.surf.gii", "*mid*.rsl.gii"]
+
+    input_dir = input_dir.absolute()
+
+    input_files = []
+    for pattern in patterns:
+        input_files.extend(input_dir.rglob(f"**/{pattern}"))
+
+    # Remove duplicates
+    return list(set(input_files))
+
+
+def detect_density(n_vertices: int) -> str:
+    density = str(round(n_vertices / 1000)) + "K"
+    return density
+
+
+def detect_hemi(structure_name: str, filename: str) -> str:
+    struct_map = {"CortexLeft": "L", "CortexRight": "R"}
+
+    if structure_name in struct_map:
+        return struct_map[structure_name]
+
+    # fallback to filename patterns
+    lower = filename.lower()
+
+    # Check for BIDS-style hemi patterns first
+    if "hemi-l" in lower:
+        return "L"
+    if "hemi-r" in lower:
+        return "R"
+
+    # Check for other common patterns
+    if "lh" in lower or "left" in lower or ".l." in lower:
+        return "L"
+    if "rh" in lower or "right" in lower or ".r." in lower:
+        return "R"
```

Comment:
```suggestion
    if any(val in lower for val in ("hemi-l", "lh", "left", ".l.")):
        return "L"
    elif any(val in lower for val in ("hemi-r", "rh", "right", ".r.")):
        return "R"
    else:
        raise Exception(f"Could not detect hemisphere. {structure_name}, {filename}")
```

I would consider using a map here, or if using conditionals, to flip it around to check if lower is in a list of words that could match the hemisphere.